source/views/stoprint/print-release.js

```diff
@@ -0,0 +1,244 @@
+			<LeftDetailCell detail="Name" title={printer.printerName} />
+			{printer.location && (
+				<LeftDetailCell detail="Location" title={printer.location} />
+			)}
+		</Section>
+	)
+}
+
+type Props = TopLevelViewPropsType & {
+	navigation: {
+		state: {params: {job: PrintJob, printer: ?Printer, username: ?string}},
+	},
+}
+
+type State = {
+	heldJob: ?HeldJob,
+	status: 'complete' | 'pending' | 'printing' | 'cancelling',
+}
+
+export class PrintJobReleaseView extends React.PureComponent<Props, State> {
+	static navigationOptions = {
+		title: 'Release Job',
+	}
+
+	state = {
+		heldJob: null,
+		status:
+			this.props.navigation.state.params.job.statusFormatted ===
+			'Pending Release'
+				? 'pending'
+				: 'complete',
+	}
+
+	componentDidMount() {
+		if (this.state.status === 'pending') {
+			this.getHeldJob()
+		}
+	}
+
+	getHeldJob = async () => {
+		const {job, printer, username} = this.props.navigation.state.params
+		const jobId = job.id.toString()
+		const response = await heldJobsAvailableAtPrinterForUser(
+			printer.printerName,
+			username,
+		)
+		if (response.error) {
+			showGeneralError(this.returnToJobsView)
+		} else {
+			const heldJobMatch = response.value.find(heldJob =>
+				heldJob.id.startsWith(jobId),
+			)
+			if (heldJobMatch) {
+				this.setState(() => ({heldJob: heldJobMatch}))
+			} else {
+				showGeneralError(this.returnToJobsView)
+			}
+		}
+	}
+
+	requestCancel = () => {
+		const {job} = this.props.navigation.state.params
+		const prompt = `Are you sure you want to cancel printing ${
+			job.documentName
+		}? This cannot be undone.`
+		Alert.alert('Print Job Cancellation Confirmation', prompt, [
+			{text: 'Cancel', style: 'cancel'},
+			{text: 'OK', onPress: this.cancelJob},
+		])
+	}
+
+	requestRelease = () => {
+		const {job, printer} = this.props.navigation.state.params
+		const prompt = `Are you sure you want to print ${job.documentName} to ${
+			printer.printerName
+		}?`
+		Alert.alert('Print Job Release Confirmation', prompt, [
+			{text: 'Cancel', style: 'cancel'},
+			{text: 'OK', onPress: this.releaseJob},
+		])
+	}
+
+	releaseJob = async () => {
+		this.setState(() => ({status: 'printing'}))
+		const {printer, username, job} = this.props.navigation.state.params
+		const {heldJob} = this.state
+		if (!heldJob) {
+			showGeneralError(this.returnToJobsView)
+			return
+		}
+		const response: ReleaseResponseOrErrorType = await releasePrintJobToPrinterForUser(
+			{
+				jobId: heldJob.id,
+				printerName: printer.printerName,
+				username: username,
+			},
+		)
+		if (response.error) {
+			Alert.alert(
+				'Error Releasing Job',
+				'We encountered a problem while trying to release your job to the printer. Please try again or release your job at the printer itself.',
+				[{text: 'OK', onPress: this.returnToJobsView}],
+			)
+		} else {
+			Alert.alert(
+				'Job Successfully Released',
+				`Document ${job.documentName} is printing at ${printer.printerName}`,
```

Comment:
Please add a period after `${printer.printerName}` for consistency with the other alert messages.